### ISSUE TITLE
Unable to update shop name, Promise.await is not a function

### ISSUE THREAD

**Issue by nnnnat** [reactioncommerce/reaction#4202]
## Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
`TypeError: Promise.await is not a function` in browser console when saving updated shop name. I believe this same issue is effecting the product handle and possibly other fields in the UI.

## Steps to Reproduce
1. Start a fresh RC shop and log in to RC as a shop admin
2. Open the shop panel, update the shop name and click the save button
3. See that the button disables
4. Close the shop panel and open it again
5. See the shop name has been reset to "Reaction"

## Possible Solution
I think this might be a regression from [PR#4049](https://github.com/reactioncommerce/reaction/pull/4049). 
Trying to lazy load the slugify/transliteration has been really tricky mainly because there are 3 versions of the same `getSlug` helper method. One that only runs on the server, one that only runs on the client and another that tries to run in both. It also seems that different parts of the UI use the different `getSLug` methods (tag nav is using `getSlug` from the `client` dir while shop name use the `getSlug` method inside of the `lib` dir).

I feel like there are two main issues to tackle
1. Creating a `getSlug` method that:
    *  takes a string as a param.
    *  can resolve/await the lazy loading of slugify/transliteration.
    *  returns the slugified string.
2. Consolidating `getSlug` usage:
    *  to a single `getSlug` method inside of `lib` that can be used by both server and client.
    *  to a server and client versions of the method and change every import of the old `lib` version.

### Versions
```sh
Node: 9.8.0
NPM: 5.8.0
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.29.0
Reaction: 1.11.0
Reaction branch: master
Docker: 18.03.0-ce
```

**Comment by tizzle** [comment]
I'm experiencing the same problem when trying to change e.g. timezone in localization/i18n settings.

**Comment by mikepuglisi** [comment]
Same Problem. Fresh install on linux mint using the reaction-cli and I can't update the store name without receiving the `Promise.await is not a function` javascript error.

**Comment by JacobGadawski** [comment]
Hi, I have same problem. I create new reaction project on fresh linux mint and I can't update store and localisation settings.

**Comment by loanlaux** [comment]
Same issue here. I have a client's multi-vendor marketplace which can't onboard new users due to that issue. As you can imagine, that's pretty critical.

Is there a quick fix one can apply while waiting for the official patch?

**Comment by nnnnat** [comment]
@loanlaux One thing you could try for a temporary fix is to remove the [dynamic import](https://github.com/reactioncommerce/reaction/blob/43e0e5dcd86444f42b910134f897ee579e9c1d2f/lib/api/helpers.js#L15) for the slugify / transliteration packages in the `lib/api/helpers.js` file and just import the one you need at the top of the file. `import { slugify } from "transliteration";` or `import slugify from "slugify";` if you're only supporting latin based chars. You'll also need to remove the `Promise.await` in the [same file](https://github.com/reactioncommerce/reaction/blob/43e0e5dcd86444f42b910134f897ee579e9c1d2f/lib/api/helpers.js#L221).

**Comment by loanlaux** [comment]
Thanks @nnnnat, I did exactly this and will stand by until it's officially fixed!

**Comment by striletskyy** [comment]
@loanlaux didn't work for us so far. Any other solutions?

**Comment by stunjiturner** [comment]
I am on the latest 1.12.0 and still cannot change shop name, from the admin

**Comment by JacobGadawski** [comment]
Same here, nothing's change in new version...

**Comment by ajporlante** [comment]
I'm closing this as this is fixed in `v1.13.0`.
